### DISCUSSION TITLE
Are there any Migration Guide for v1.0.0?

### DISCUSSION THREAD

**Discussion by ghiscoding** [axios/axios#4996] [General]
I'm really excited about this new version 1.0.0 that just came out but I cannot find any Migration Guide anywhere. All I could find is the following in the changelog

> There are multiple deprecations, refactors and fixes provided in this release. Please read through the full release notes to see how this may impact your project and use case.

But that is not very helpful, I don't wish to have to go through every single PR that came out to find what needs to change. Are there any Migration Guide here or over the internet that can be helpful? 

Thanks for the new version

**Comment by jasonsaayman** [comment]
I will be adding one as soon as possible.

**Reply by bombillazo** [reply]
Any update on this? Went searching online and wasn't able to find one.

**Reply by ishaiavrahami** [reply]
?

**Reply by baraeb92** [reply]
Upgrading to v1 break our app, and without official migration guide, it is much harder to convince the team to do it, because there is so much unknown around what will break.

**Reply by LucasNovaes2016** [reply]
@jasonsaayman Any update on the migration guide?

**Reply by ibeladi-mrsool** [reply]
A partial work-in-progress migration guide that resolves main issues will be much better than waiting for a fully comprehensive migration guide that will take long time. I would assume that most of the projects uses axios in a very shallow level, so these projects would benefit from any migration guide that covers most used cases.

**Reply by doutatsu** [reply]
Bumping this up again - would love to upgrade, considering we already on 1.2 - yet no migration guide exists...

**Reply by amirhamdy** [reply]
any update on this would be really appreciated.

**Reply by brandonlenz** [reply]
Agree, a migration guide would be awesome. Is there at least a list of the breaking changes so that consumers will know what to look out for, and gauge the degree to which we might be affected?

e.g. the [1.0.0 tag does not list which changes are breaking](https://github.com/axios/axios/releases/tag/v1.0.0)

**Reply by PreisM** [reply]
Guide would be great, any update on this?

**Reply by MaximeAnsquer** [reply]
Hello, an upgrade guide would be very useful for me too. Thanks in advance!

**Reply by TomasTarasovic** [reply]
It has been over a year, is there any update about the upgrade guide? Thanks!

**Reply by pavelrahman619** [reply]
Still no update?

**Reply by quentingrchr** [reply]
Still no updates ?

**Reply by bcass** [reply]
Looks like it's here! https://github.com/axios/axios/blob/v1.x/MIGRATION_GUIDE.md

**Comment by LucasNovaes2016** [comment]
It would be really hepful if there was a migration guide.

**Reply by lODIN007l** [reply]
I think little by little they are considering making the guide

**Comment by jasonsaayman** [comment]
Yeah I will get to finishing it off, sorry been a busy time for me with IRL work

**Reply by amirhamdy** [reply]
any updates on this?

**Reply by baraeb92** [reply]
Most likely landing after a new year, I supposed. 

It's the holiday season, so the maintainer understandably needed to take some time away from their primary job and their open-source project

**Reply by melyux** [reply]
2 years later...

**Comment by tmcdos** [comment]


**Reply by plourenco** [reply]
Is there any gain from comments like this? I've just subscribed to this thread to get feedback about the states of the migration guide. But these kind of comments don't speed up things, they just spam people...

**Comment by ames20** [comment]
I had issues upgrading as well - any progress on the guide?

**Comment by cburkins** [comment]
Agree with others, having a few bullet points on common errors during migration to v1.0 would be better than waiting for a perfect migration guide.

**Reply by miguelpinia** [reply]
I agree with this; it would be useful to keep updated our applications. Meanwhile, I won't touch anything related to axios.

**Comment by ghiscoding** [comment]
hey guys, so I'm the one who opened this discussion but since I'm still getting a lot of notifications that a migration guide is needed... We did the move a few months ago and it was actually pretty easy to do the migration. Our project uses Vue 3 and Vite, so prior to Axios 1.x being released, we were using [@bundled-es-modules/axios](https://www.npmjs.com/package/@bundled-es-modules/axios) to get Axios 0.x working as ESM in Vite, and the migration from that temp lib to the new Axios 1.x was quite simple

```diff
- import { axios } from '@bundled-es-modules/axios';
+ import axios from 'axios';
```
the only other issue we had was with the `AxiosRequestConfig` interface that we use in our http interceptors and for that we simply had to switch to the `InternalAxiosRequestConfig` interface 

```diff
- import { axios, AxiosRequestConfig, AxiosResponse } from '@bundled-es-modules/axios';
+ import axios, { AxiosError, AxiosResponse, InternalAxiosRequestConfig } from 'axios';

function startInterceptors() {
  axios.interceptors.request.use(onRequestFullfilled, onRequestResponseRejected);
  axios.interceptors.response.use(onResponseFullfilled, onRequestResponseRejected);
}

async function onRequestResponseRejected(reason: AxiosError<string>) {
-  const originalRequest = reason.config;
+  const originalRequest = reason.config as InternalAxiosRequestConfig;
   // ...
}
```
and that's about it... obviously this doesn't cover all use case, but it does cover the basic usage. So there's no need to wait for a perfect Migration Guide, just give a try ;)

Cheers and happy coding

**Reply by bmuenzenmeyer** [reply]
inspired by this post, I've quick created https://github.com/bmuenzenmeyer/axios-1.0.0-migration-guide/

- we as a community of 99k stars and 40+ million weekly downloads have the collective wisdom to track these changes, spread the word, add what you can! I'll be encouraging my team to do so.
- @ghiscoding I've borrowed your diffs (hope you are okay with that) - thanks for getting the ball rolling
- (no warranties, but I hope it helps some folks until something official lands)

**Reply by josias-r** [reply]
We have an interceptor to fetch a new auth token when the old one was outdated and then update the request headers.
This also slightly changed in v1:

```diff
-    if (request.headers?.common?.Authorization) {
-      request.headers.common.Authorization =
+    if (request.headers?.Authorization) {
+      request.headers.Authorization =
        axiosInstance.defaults.headers.common.Authorization;
    }
```

In general v1 has specfic Header types now, and it seems it reworked headers overall, so be careful if you have custom implementations working with headers in some way

**Reply by vovkvlad** [reply]
One more thing I encountered: `paramsSerializer` structure has changed. Although the new README has a new structure, the official docs site is still referring to the old structure. 
So if you have code like this in your requests:

```diff
axios.get(url, {
- paramsSerializer: (params) =>
-           doSmth(params),
+ paramsSerializer: { serialize: (params) => doSmth }
```

**Reply by jasonsaayman** [reply]
Thanks, I really will still get this out, I know it's largely on me and I have started. I will look to put up what I have thus far this weekend and open for additions from the community

**Reply by vovkvlad** [reply]
jsut in case: I've opened a PR updating `req_config.md` adding new stuff that was added recently, and updating existing props that has been changed : https://github.com/axios/axios-docs/pull/126

**Reply by jasonsaayman** [reply]
awesome thanks

**Reply by rago4** [reply]
Your solution saved me hours of work, thank you! 🙇‍♂️

**Reply by ericyd** [reply]
This might have been true in the past but I believe axios now supports either an object with `serialize` property or a function https://github.com/axios/axios/blob/b3be36585884ba1e237fdd0eacf55f678aefc396/lib/core/Axios.js#L61-L70

**Comment by rfgonzalezweb** [comment]
hi, is there any action required to migrate from 0.24.X -> 1.4.0 ?

I checked this guide but couldn't find any clear path.
https://github.com/axios/axios/blob/main/UPGRADE_GUIDE.md

thanks in advance

**Comment by leonbloy** [comment]
See also

https://github.com/bmuenzenmeyer/axios-1.0.0-migration-guide

https://github.com/bmuenzenmeyer/axios-1.0.0-migration-guide/issues

**Comment by MarwenLabidi** [comment]
any update on this ?
i had problem when i migrate axios from **[0.27.2]** to **[1.0.0]**
all the ws request return **unauthorized**  status?

**Reply by tameroski** [reply]
Had a similar issue where all my requests were missing the "Cookie" header, resulting in an unauthorized status. Adding the `withCredentials: true` parameter to each request solved the problem.
Might also be a server configuration issue (see https://github.com/axios/axios/issues/5824), but didn't tried that.

**Comment by ac185422** [comment]
Hi, just curious if there is any update on the migration guide and if latest 1.x version (currently 1.6.7) is stable to upgrade from 0.28.0 to. I see the latest 0.x version has fixed vulnerabilities in 0.27.2 as well as backported some 1.x features however I see others have held out on going above 1.x due to issues listed above as well as another post in which I read some 1.x versions were not stable https://github.com/axios/axios/issues/5114#issuecomment-1277169905. Any guidance on this would be greatly appreciated

**Reply by KrisLau** [reply]
Nope still none thus far. For now you'll have to rely on [external guides](https://github.com/bmuenzenmeyer/axios-1.0.0-migration-guide) or just stick with 0.27 from what I can tell. I'll probs stick with 0.27 for now, been checking back every couple of months to look for a migration guide haha

**Reply by melyux** [reply]
Tried updating past 0.27, and ran into random breaking errors which weren't documented anywhere. Back to 0.27. Might get rid of Axios and use plain fetch() now that it's caught up to Axios.

**Reply by KrisLau** [reply]
@melyux You can try [ky](https://github.com/sindresorhus/ky) if you were thinking of using fetch. It's been recommended in a bunch of forums and from what I can tell, seems to be well regarded

**Comment by jasonsaayman** [comment]
going to close as we now have a guide in the repo
